### PR TITLE
ARROW-15501: [Java] Support validating decimal vectors

### PR DESCRIPTION
**Pull request by liyafan82**
Support validating decimal vectors and check precisions.

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/validate/ValidateVectorTypeVisitor.java [hunk 1]
```diff
@@ -116,6 +116,25 @@ private void validateDateVector(ValueVector vector, DateUnit expectedDateUnit) {
         "Expecting date unit %s, actual date unit %s.", expectedDateUnit, dateType.getUnit());
   }
 
+  private void validateDecimalVector(ValueVector vector) {
+    validateOrThrow(vector.getField().getFieldType().getType() instanceof ArrowType.Decimal,
+            "Vector %s is not a decimal vector", vector.getClass());
+    ArrowType.Decimal decimalType = (ArrowType.Decimal) vector.getField().getFieldType().getType();
+    validateOrThrow(decimalType.getScale() >= 0, "The scale of decimal %s is negative.", decimalType.getScale());
```
Comment (emkornfield) [inline java/vector/src/main/java/org/apache/arrow/vector/validate/ValidateVectorTypeVisitor.java:123]: we should also validate precision is less then scale.

**Comment by emkornfield** [comment]
Nice catch, i think one more check should be done.  How was this caught?

File: java/vector/src/main/java/org/apache/arrow/vector/validate/ValidateVectorTypeVisitor.java (hunk 1 above)
Comment (liyafan82) [reply java/vector/src/main/java/org/apache/arrow/vector/validate/ValidateVectorTypeVisitor.java:123]: Good point. Thanks. 
Added a check for that and a test case for it.

**Comment by liyafan82** [comment]
> Nice catch, i think one more check should be done. How was this caught?

Thanks for your feedback. 
It was caught by previous (C++) fuzzing files, originally submitted in another PR. Please see (https://github.com/apache/arrow/pull/10789/files)

**Comment by liyafan82** [comment]
@emkornfield Thanks for your comments. If you have no more comment, may I merge it?

**Comment by emkornfield** [comment]
Sorry, please go ahead an merge.

**Pull request closed without merging**